.golangci.yml

```diff
@@ -10,36 +11,35 @@ linters:
     # - funlen
     # - gochecknoglobals
     # - gochecknoinits
+    # - gocognit
     - goconst
-    - gocritic
+    # - gocritic
     # - gocyclo
     # - godox
     - gofmt
     - goimports
-    - golint
+    - revive
     - gosec
     - gosimple
     - govet
     - ineffassign
     # - interfacer
     - lll
-    - misspell
     # - maligned
+    # - misspell
     - nakedret
+    - nolintlint
     - prealloc
-    # - scopelint
     - staticcheck
     - structcheck
     - stylecheck
-    - typecheck
+    # - typecheck
```

Comment:
?

Reply:
this was pulled from upstream https://github.com/celestiaorg/celestia-core/pull/637/commits/5d806709a9dbfed4f7dda2facb49d46604584d36

Reply:
I think was originally removed by upstream to debug many type check errors caused by the linter auto upgrading to the latest version of go, but after forcing 1.17, it is not needed. not 100% positive, but I didn't get any errors when adding it back in

https://github.com/celestiaorg/celestia-core/pull/637/commits/4aa56e021309e70f7267e258eecab11d2bf30cf3